systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java

```diff
@@ -673,6 +673,16 @@ public static void waitForKafkaTopicPartitionChange(String topicName, int partit
         );
     }
 
+    public static void waitForKafkaTopicReplicationFactorChange(String topicName, int replicationFactor) throws InterruptedException {
+        LOGGER.info("Waiting for Kafka topic change {}", topicName);
+        TestUtils.waitFor("Waits for Kafka topic change " + topicName, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS,
+                Constants.TIMEOUT_FOR_RESOURCE_READINESS, () ->
+                        Crds.topicOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace())
+                                .withName(topicName).get().getSpec().getReplicas() == replicationFactor
```

Comment:
Shouldn't this use the CR Status? Otherwise this is vulnerable to timing - the change of replicas in the CR doesn't mean the TO saw it already.